current trunk does not work

```
2014-07-01T23:30:42.435-0700    ERROR   main    com.facebook.presto.server.PrestoServer Guice creation errors:

1) Error in custom provider, java.lang.IllegalStateException: Jax-rs services must be explicitly bound using the JaxRsBinder:  [[Key[type=io.airlift.discovery.server.DynamicAnnouncementResource, annotation=[none]], Key[type=io.airlift.discovery.server.ServiceResource, annotation=[none]]]]
  at io.airlift.jaxrs.JaxrsModule.createResourceConfig(JaxrsModule.java:86)
  while locating com.sun.jersey.api.core.ResourceConfig
  at io.airlift.jaxrs.JaxrsModule.createServletContainer(JaxrsModule.java:79)
  while locating com.sun.jersey.spi.container.servlet.ServletContainer
  while locating javax.servlet.Servlet annotated with @io.airlift.http.server.TheServlet()
    for parameter 3 at io.airlift.http.server.HttpServerProvider.<init>(HttpServerProvider.java:70)
  while locating io.airlift.http.server.HttpServerProvider
  at io.airlift.http.server.HttpServerModule.configure(HttpServerModule.java:61)
  while locating io.airlift.http.server.HttpServer

1 error
com.google.inject.CreationException: Guice creation errors:

1) Error in custom provider, java.lang.IllegalStateException: Jax-rs services must be explicitly bound using the JaxRsBinder:  [[Key[type=io.airlift.discovery.server.DynamicAnnouncementResource, annotation=[none]], Key[type=io.airlift.discovery.server.ServiceResource, annotation=[none]]]]
  at io.airlift.jaxrs.JaxrsModule.createResourceConfig(JaxrsModule.java:86)
  while locating com.sun.jersey.api.core.ResourceConfig
  at io.airlift.jaxrs.JaxrsModule.createServletContainer(JaxrsModule.java:79)
  while locating com.sun.jersey.spi.container.servlet.ServletContainer
  while locating javax.servlet.Servlet annotated with @io.airlift.http.server.TheServlet()
    for parameter 3 at io.airlift.http.server.HttpServerProvider.<init>(HttpServerProvider.java:70)
  while locating io.airlift.http.server.HttpServerProvider
  at io.airlift.http.server.HttpServerModule.configure(HttpServerModule.java:61)
  while locating io.airlift.http.server.HttpServer

1 error
    at com.google.inject.internal.Errors.throwCreationExceptionIfErrorsExist(Errors.java:435) ~[guice-3.0.jar:na]
    at com.google.inject.internal.InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:183) ~[guice-3.0.jar:na]
    at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:109) ~[guice-3.0.jar:na]
    at com.google.inject.Guice.createInjector(Guice.java:95) ~[guice-3.0.jar:na]
    at io.airlift.bootstrap.Bootstrap.initialize(Bootstrap.java:286) ~[bootstrap-0.92-SNAPSHOT.jar:0.92-SNAPSHOT]
    at com.facebook.presto.server.PrestoServer.run(PrestoServer.java:105) [presto-main-0.74-SNAPSHOT.jar:0.74-SNAPSHOT]
    at com.facebook.presto.server.PrestoServer.main(PrestoServer.java:61) [presto-main-0.74-SNAPSHOT.jar:0.74-SNAPSHOT]
Caused by: java.lang.IllegalStateException: Jax-rs services must be explicitly bound using the JaxRsBinder:  [[Key[type=io.airlift.discovery.server.DynamicAnnouncementResource, annotation=[none]], Key[type=io.airlift.discovery.server.ServiceResource, annotation=[none]]]]
    at com.google.common.base.Preconditions.checkState(Preconditions.java:200) ~[guava-16.0.1.jar:na]
    at io.airlift.jaxrs.JaxrsModule.createResourceConfig(JaxrsModule.java:105) ~[jaxrs-0.92-SNAPSHOT.jar:0.92-SNAPSHOT]
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.7.0_55]
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[na:1.7.0_55]
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_55]
    at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_55]
    at com.google.inject.internal.ProviderMethod.get(ProviderMethod.java:104) ~[guice-3.0.jar:na]
    at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:40) ~[guice-3.0.jar:na]
    at com.google.inject.internal.InjectorImpl$4$1.call(InjectorImpl.java:978) ~[guice-3.0.jar:na]
    at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1031) ~[guice-3.0.jar:na]
    at com.google.inject.internal.InjectorImpl$4.get(InjectorImpl.java:974) ~[guice-3.0.jar:na]
    at com.google.inject.spi.ProviderLookup$1.get(ProviderLookup.java:89) ~[guice-3.0.jar:na]
    at com.google.inject.internal.ProviderMethod.get(ProviderMethod.java:98) ~[guice-3.0.jar:na]
    at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:40) ~[guice-3.0.jar:na]
    at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:54) ~[guice-3.0.jar:na]
    at com.google.inject.internal.SingleParameterInjector.inject(SingleParameterInjector.java:38) ~[guice-3.0.jar:na]
    at com.google.inject.internal.SingleParameterInjector.getAll(SingleParameterInjector.java:62) ~[guice-3.0.jar:na]
    at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:84) ~[guice-3.0.jar:na]
    at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:254) ~[guice-3.0.jar:na]
    at com.google.inject.internal.BoundProviderFactory.get(BoundProviderFactory.java:53) ~[guice-3.0.jar:na]
    at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46) ~[guice-3.0.jar:na]
    at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1031) ~[guice-3.0.jar:na]
    at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40) ~[guice-3.0.jar:na]
    at com.google.inject.Scopes$1$1.get(Scopes.java:65) ~[guice-3.0.jar:na]
    at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:40) ~[guice-3.0.jar:na]
    at com.google.inject.internal.InternalInjectorCreator$1.call(InternalInjectorCreator.java:204) ~[guice-3.0.jar:na]
    at com.google.inject.internal.InternalInjectorCreator$1.call(InternalInjectorCreator.java:198) ~[guice-3.0.jar:na]
    at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1024) ~[guice-3.0.jar:na]
    at com.google.inject.internal.InternalInjectorCreator.loadEagerSingletons(InternalInjectorCreator.java:198) ~[guice-3.0.jar:na]
    at com.google.inject.internal.InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:179) ~[guice-3.0.jar:na]
    ... 5 common frames omitted
```

--- a/presto-main/src/main/java/com/facebook/presto/discovery/EmbeddedDiscoveryModule.java
+++ b/presto-main/src/main/java/com/facebook/presto/discovery/EmbeddedDiscoveryModule.java
@@ -42,6 +42,7 @@ import java.util.Set;
 import static com.google.common.base.Preconditions.checkNotNull;
 import static io.airlift.configuration.ConfigurationModule.bindConfig;
 import static io.airlift.discovery.client.DiscoveryBinder.discoveryBinder;
+import static io.airlift.jaxrs.JaxrsBinder.jaxrsBinder;
 import static io.airlift.json.JsonCodecBinder.jsonCodecBinder;
 
 public class EmbeddedDiscoveryModule
@@ -55,7 +56,7 @@ public class EmbeddedDiscoveryModule
         }
 
         bindConfig(binder).to(DiscoveryConfig.class);
-        binder.bind(ServiceResource.class).in(Scopes.SINGLETON);
+        jaxrsBinder(binder).bind(ServiceResource.class);
 
         discoveryBinder(binder).bindHttpAnnouncement("discovery");
 
@@ -65,7 +66,7 @@ public class EmbeddedDiscoveryModule
         binder.bind(ServiceSelector.class).to(DiscoveryServiceSelector.class);
         binder.bind(StaticStore.class).to(EmptyStaticStore.class);
 
-        binder.bind(DynamicAnnouncementResource.class).in(Scopes.SINGLETON);
+        jaxrsBinder(binder).bind(DynamicAnnouncementResource.class);
         binder.bind(DynamicStore.class).to(ReplicatedDynamicStore.class).in(Scopes.SINGLETON);
         binder.install(new ReplicatedStoreModule("dynamic", ForDynamicStore.class, InMemoryStore.class));
     }

